src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -126,7 +126,7 @@ private BindingDataResolver GetResolver(PropertyInfo propInfo, INameResolver nam
             // first try to resolve with connection string
             if (connStrAttr != null)
             {
-                return GetConfigurationResolver(str, connStrAttr.Default, propInfo, validator, s => _configuration.GetConnectionStringOrSetting(s));
+                return GetConfigurationResolver(str, connStrAttr.Default, propInfo, validator, s => _configuration.GetConnectionStringOrSetting(nameResolver.Resolve(s)));
```

Comment:
Need to add tests. I believe there is already an attribute cloner test suite you can add to